website/parser_test.ts

```diff
@@ -0,0 +1,458 @@
+  const X = doc.filter(
+    x => x.type === "VariableDeclarationList" && x.declarations[0].name === "X"
+  );
+  assertEqual(X.length, 1);
+  assertEqual(X[0].declarations[0].name, "X");
+  assertEqual(X[0].isPrivate, true);
+  const P = doc.filter(
+    p => p.type === "VariableDeclarationList" && p.declarations[0].name === "P"
+  );
+  assertEqual(P.length, 1);
+  assertEqual(P[0].declarations[0].name, "P");
+  assertEqual(P[0].isPrivate, true);
+});
+
+test(async function test_js() {
+  const doc = generateDoc("testdata/file.js", options);
+  assertEqual(doc.length, 4);
+  const fooExport = doc.filter(x => x.name === "foo" && x.type === "export");
+  assertEqual(fooExport.length, 1);
+  const fooFn = doc.filter(x => x.name === "foo" && x.type === "function");
+  assertEqual(fooFn.length, 1);
+  const defaultExport = doc.filter(x => x.type === "export" && x.isDefault);
+  assertEqual(defaultExport.length, 1);
+  assertEqual(defaultExport[0].propertyName, "defaultExport");
+  const fn = doc.filter(
+    x => x.type === "function" && x.name === "defaultExport"
+  );
+  assertEqual(fn.length, 1);
+});
+
+test(async function test_class() {
+  const doc = generateDoc("testdata/class.ts", options);
+  assertEqual(doc.length, 1);
+  const C1 = doc[0];
+  assertEqual(C1.type, "class");
+  assertEqual(C1.name, "Point");
+  assertEqual(C1.isAbstract, false);
+  assertEqual(C1.implementsClauses.length, 1);
+  assertEqual(C1.implementsClauses[0].filename, "./point");
+  assertEqual(C1.members.length, 4);
+  // Test Constructor
+  assertEqual(C1.members[0].type, "Constructor");
+  assertEqual(C1.members[0].parameters.length, 3);
+  assertEqual(C1.members[0].parameters[0].name, "x");
+  assertEqual(C1.members[0].parameters[0].visibility, "public");
+  // Test distance()
+  assertEqual(C1.members[1].type, "MethodDeclaration");
+  assertEqual(C1.members[1].name, "distance");
+  assertEqual(C1.members[1].parameters.length, 1);
+  assertEqual(C1.members[1].parameters[0].dataType.name, "types.Point");
+  assertEqual(C1.members[1].parameters[0].dataType.filename, "./point");
+  assertEqual(C1.members[1].returnType.type, "keyword");
+  assertEqual(C1.members[1].returnType.name, "number");
+  // Test square()
+  assertEqual(C1.members[2].type, "MethodDeclaration");
+  assertEqual(C1.members[2].name, "square");
+  assertEqual(C1.members[2].visibility, "private");
+  // Test isUnderXAxis()
+  assertEqual(C1.members[3].type, "MethodDeclaration");
+  assertEqual(C1.members[3].name, "isUnderXAxis");
+  assertEqual(C1.members[3].isStatic, true);
+});
+
+test(async function test_interface() {
+  const doc = generateDoc("testdata/point.ts", options);
+  // Test Point
+  const Point = doc[0];
+  assertEqual(Point.name, "Point");
+  assertEqual(Point.type, "interface");
+  assertEqual(Point.documentation, "Represents a point in 3D space.");
+  assertEqual(Point.members.length, 3);
+  assertEqual(Point.parameters.length, 0);
+  assertEqual(Point.members[0].name, "x");
```

Comment:
Check that you get the documentation string "Represents x value of a point."

Reply:
done